Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -5045,6 +5045,10 @@ void WebPage::updateRendering()
 {
     protectedCorePage()->updateRendering();
 
+#if PLATFORM(WPE)
+    flushPendingThemeColorChange();
+#endif
+
```

Comment:
Why do we need this here? It's already called by the drawing area right after updateRedering().

Reply:
I left the second place in for review, this one is the one that works, the one triggered from the drawing area doesn't. Should I remove it from the drawing area?

Reply:
Left the intact one in there, removed the one from the drawing area. I suppose if this gets ported to GTK you might want to find a better spot to put the flush operation.

Reply:
Also, unless you understand why this is not needed for macOS, the guard should be:

```
#if PLATFORM(MAC) || PLATFORM(WPE)
```

Reply:
Well, I see the reason: macOS does this in `TiledCoreAnimationDrawingArea::updateRendering`. Ideally we would parallel that to reduce odds of platform-specific bugs, but I see you've already tried....